articles/google-play-in-app-reviews/index.md

```diff
@@ -0,0 +1,250 @@
+
+### Final Activity Code
+***Remember to press `alt + enter` on a PC and `option + enter` on a Mac to import the classes after copying and pasting the code blocks into your IDE.***
+
+```java
+public class MainActivity extends AppCompatActivity {
+    ReviewManager reviewManager;
+    ReviewInfo reviewlnfo = null;
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        getReviewInfo();
+        startReviewFlow();
+    }
+
+    private void getReviewInfo() {
+        reviewManager = ReviewManagerFactory.create(getApplicationContext());
+        Task<ReviewInfo> manager = reviewManager.requestReviewFlow();
+        manager.addOnCompleteListener(task -> {
+            if (task.isSuccessful()) {
+                reviewlnfo = task.getResult();
+            } else {
+                Toast.makeText(getApplicationContext(), "In App ReviewFlow failed to start", Toast.LENGTH_LONG).show();
+            }
+        });
+    }
+
+    public void startReviewFlow() {
+        if (reviewlnfo != null) {
+            Task<Void> flow = reviewManager.launchReviewFlow(this, reviewlnfo);
+            flow.addOnCompleteListener(new OnCompleteListener<Void>() {
+                @Override
+                public void onComplete(Task<Void> task) {
+                    Toast.makeText(getApplicationContext(), "In App Rating complete", Toast.LENGTH_LONG).show();
+                }
+            });
+        }
+        else {
+            Toast.makeText(getApplicationContext(), "In App Rating failed", Toast.LENGTH_LONG).show();
+        }
+    }
+}
+```
+
+### Testing
+To test our application, we will use internal app sharing. To confirm that this works, add a `Button` to trigger the review flow. Remember, we are using a call to action for testing. In production applications, calls to action such as `Button` should be avoided when using this API.
+
+- Add a button in your activity.xml
+
+```xml
+<Button
+    android:id="@+id/button"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="159dp"
+    android:layout_marginEnd="164dp"
+    android:layout_marginBottom="600dp"
+    android:text="SHOW IN APP REVIEW DIALOG"
+    app:layout_constraintBottom_toBottomOf="parent"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent" />
+```
+
+- Declare the button
+
+```java
+Button mButton;
+```
+
+Below `onCreate`,
+
+- Create an instance of the `mButton`
+
+```java
+mButton = findViewById(R.id.button);
+```
+
+- Add `OnClickListener` and call `startReviewFlow()`
+
+```java
+mButton.setOnClickListener(new View.OnClickListener() {
+    @Override
+    public void onClick(View view) {
+        startReviewFlow();
+    }
+});
+```
+
+#### What you need
```

Comment:
I think this should have been added at the prerequisites at the beginning of the article to prevent a reader from going further into the article only to find there are more requirements.